### PR TITLE
Update agent monitoring config to support /inputs of filebeat

### PR DESCRIPTION
**Pull request by kcreddy**
## What does this PR do?

- Adds agent configuration to monitor filebeat's `/inputs/` endpoint.
Reference: https://github.com/elastic/beats/issues/33953#issuecomment-1371349692

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)

## Related issues
- Relates https://github.com/elastic/beats/issues/33953
- Relates https://github.com/elastic/security-team/issues/5525

### CODE REVIEW

File: internal/pkg/agent/application/monitoring/inputs.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	"net/http"
+
+	"github.com/elastic/elastic-agent-libs/monitoring"
+)
+
+func inputsHandler(ns *monitoring.Namespace) func(http.ResponseWriter, *http.Request) error {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
+
+		data := monitoring.CollectStructSnapshot(
```
Comment (michalpristas) [inline internal/pkg/agent/application/monitoring/inputs.go:19]: this should be exposed in filebeat right? why are we adding inputs handler here?

File: internal/pkg/agent/application/monitoring/v1_monitor.go [hunk 2]
```diff
@@ -714,6 +714,69 @@ func (b *BeatsMonitor) injectMetricsInput(cfg map[string]interface{}, componentI
...
+					"dataset":   fmt.Sprintf("elastic_agent.%s", fbDataStreamName),
+					"namespace": monitoringNamespace,
+				},
+				"metricsets":    []interface{}{"json"},
+				"hosts":         endpoints,
+				"path":          "/inputs/",
+				"namespace":     fbDataStreamName,
+				"json.is_array": true,
+				"period":        "10s",
+				"index":         fmt.Sprintf("metrics-elastic_agent.%s-%s", fbDataStreamName, monitoringNamespace),
```
Comment (michalpristas) [inline internal/pkg/agent/application/monitoring/v1_monitor.go:733]: make sure that we have permissions to write to this index. 
we have a limited output permissions (part of the incoming policy `output_permissions`)

Comment (kcreddy) [reply internal/pkg/agent/application/monitoring/v1_monitor.go:733]: The permission to auto generate `filebeat_input` indices is in this merged PR: https://github.com/elastic/kibana/pull/149974

File: internal/pkg/agent/application/monitoring/inputs.go (hunk 1 above)
Comment (kcreddy) [reply internal/pkg/agent/application/monitoring/inputs.go]: I removed this inputs handler.

File: changelog/fragments/1674822835-Enhance-agent-monitoring-config-to-support-filebeat-inputs-endpoint [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# Change summary; a 80ish characters long description of the change.
+summary: Enhance agent monitoring config to support filebeat /inputs endpoint
+
+# Long description; in case the summary is not enough to describe the change
+# this field accommodate a description without length limits.
+# NOTE: This field will be rendered only for breaking-change and known-issue kinds at the moment.
+#description:
+
+# Affected component; a word indicating the component this changeset affects.
+component: filebeat
```
Comment (AndersonQ) [inline changelog/fragments/1674822835-Enhance-agent-monitoring-config-to-support-filebeat-inputs-endpoint:22]: [Blocker]
Filebeat isn't really a component in this context. Better would me `monitoring`

**Review by AndersonQ — CHANGES_REQUESTED** [review]

**Review by michalpristas — APPROVED** [review]
lgtm now

File: changelog/fragments/1674822835-Enhance-agent-monitoring-config-to-support-filebeat-inputs-endpoint (hunk 3 above)
Comment (kcreddy) [reply changelog/fragments/1674822835-Enhance-agent-monitoring-config-to-support-filebeat-inputs-endpoint]: updated `component` to monitoring.

**Review by AndersonQ — APPROVED** [review]

**Pull request merged**